File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/dagger/modules/UpdateSchedulerModule.java
package se.gustavkarlsson.aurora_notifier.android.dagger.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import se.gustavkarlsson.aurora_notifier.android.background.UpdateScheduler;
import se.gustavkarlsson.aurora_notifier.android.settings.Settings;

@Module
public abstract class UpdateSchedulerModule {

	@Provides
	@Singleton
	static UpdateScheduler provideUpdateScheduler(Context context, Settings settings) {
		return new UpdateScheduler(context, settings);
	}

}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/modules/UpdateSchedulerModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules

import dagger.Module
import dagger.Provides
import org.threeten.bp.Duration
import se.gustavkarlsson.skylight.android.services.Scheduler
import se.gustavkarlsson.skylight.android.services_impl.scheduling.GetLatestAuroraReportScheduler
import javax.inject.Singleton

@Module
class UpdateSchedulerModule {

	// Published
	@Provides
	@Singleton
	fun provideUpdateScheduler(): Scheduler =
		GetLatestAuroraReportScheduler(SCHEDULE_INTERVAL, SCHEDULE_FLEX)

	companion object {
		private val SCHEDULE_INTERVAL: Duration = Duration.ofMinutes(20)
		private val SCHEDULE_FLEX: Duration = Duration.ofMinutes(10)
	}
}
